internal/service/common/repo/repository.go

```diff
@@ -84,3 +89,111 @@ func (r *CommonRepository) DeleteDataChangeEvent(ctx context.Context, id uuid.UU
 func (r *CommonRepository) GetDataChangeEvents(ctx context.Context) ([]models2.DataChangeEvent, error) {
 	return utils.FindAll[models2.DataChangeEvent](ctx, r.Db)
 }
+
+// ClaimDataChangeEvent claims a batch of DataChangeEvent and updates the high-water mark.
+func ClaimDataChangeEvent(pool *pgxpool.Pool, ctx context.Context) ([]models2.DataChangeEvent, error) {
```

Comment:
Would you mind renaming the `models2` import to `models` while you are in here.  Must have gotten messed up when the code was moved around.

Reply:
renamed to `commonmodels` (just `models` was causing an error when generating the mock file)